library/src/include/utility.h

```diff
@@ -346,4 +347,25 @@ inline std::ostream& operator<<(std::ostream& os, rocblas_half x)
 {
     return os << float(x);
 }
+
+// Convert the current C++ exception to rocblas_status
+// This allows extern "C" functions to return this function in a catch(...) block
+// while converting all C++ exceptions to an equivalent rocblas_status here
+inline rocblas_status exception_to_rocblas_status()
+try
+{
+    auto e = std::current_exception();
+    if(e)
+        std::rethrow_exception(e);
+    return rocblas_status_success;
+}
+catch(std::bad_alloc)
+{
+    return rocblas_status_memory_error;
+}
+catch(...)
+{
+    return rocblas_status_internal_error;
```

Comment:
Is it really necessary to remove curly brackets?

Reply:
We are not _removing_ curly brackets.

If you're asking whether they are needed around a 1-line `catch` handler, [yes, they are](https://en.cppreference.com/w/cpp/language/try_catch). A `try` / `catch` block requires _compound statements_ for the `try` and `catch` clauses, which are brace-enclosed.

Reply:
You wrote:
void foo() try {} catch(..) {}

I am saying why not writing it like:
void foo() { try {} catch(..) {} }

Reply:
> 
> 
> You wrote:
> void foo() try {} catch(..) {}
> 
> I am saying why not writing it like:
> void foo() { try {} catch(..) {} }

Because the former is _function_ try-block, and thus it does not interfere with the indentation of the code -- all we're saying is _"try this function, exactly as it was already written, and if it raises any exceptions, handle them in blocks after the function. You do not need to reindent your code, and the extra code for handling exceptions will be at the end of the function and thus not a distraction from the main part of the code."_

[Function try blocks](https://en.cppreference.com/w/cpp/language/function-try-block) are mainly for constructors, since there's no other way to catch exceptions thrown during base or member initialization lists (by the time you're in the main body of the constructor, any exceptions raised during initialization would have escaped). They are also good for this use-case, where we want to stop any exceptions from escaping our C API, but we don't write our code thinking mainly about the exceptions, or organizing our code indentation around them.